### PR TITLE
Tests for multiple investors in same fund + investment in different assets 

### PR DESCRIPTION
**Pull request by jennazenk**
- requestedShares argument is not optional anymore, it is required (in requestInvestment and invest) 

@schmidsi @gempi

### CODE REVIEW

File: src/contracts/fund/hub/transactions/setupFund.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
+import * as R from 'ramda';
```
Comment (schmidsi) [inline src/contracts/fund/hub/transactions/setupFund.ts]: I think this is a test-only file? Can you please move it to `src/tests/utils`? There are already other files like this.

File: src/contracts/fund/participation/transactions/executeRequestFor.ts [hunk 2]
```diff
@@ -30,9 +31,7 @@ const guard = async (environment, params, contractAddress) => {
     'Amount of requested shares is null',
   );
 
-  // TODO: remaining pre flights
-  // ensure isRecent
-  // ensure isPriceRecent
+  await ensureHasValidRequest(environment, contractAddress, params.who);
```
Comment (schmidsi) [inline src/contracts/fund/participation/transactions/executeRequestFor.ts]: 👍

File: src/contracts/fund/participation/transactions/invest.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { getRoutes } from '~/contracts/fund/hub/calls/getRoutes';
```
Comment (schmidsi) [inline src/contracts/fund/participation/transactions/invest.ts]: I think this is a test-only file? Can you please move it to src/tests/utils? There are already other files like this.

File: src/contracts/fund/hub/transactions/setupFund.ts (hunk 1 above)
Comment (jennazenk) [reply src/contracts/fund/hub/transactions/setupFund.ts]: no its not a test file ; it allows to create a fund without investing in it

File: src/contracts/fund/participation/transactions/invest.ts (hunk 3 above)
Comment (jennazenk) [reply src/contracts/fund/participation/transactions/invest.ts]: same as setupFund

File: src/contracts/fund/hub/transactions/setupFund.ts (hunk 1 above)
Comment (jennazenk) [reply src/contracts/fund/hub/transactions/setupFund.ts]: Can be used by bots / Midas as a way to easily spin up a fund

Comment (schmidsi) [reply src/contracts/fund/hub/transactions/setupFund.ts]: True.

**Pull request merged**